src/main/kotlin/org/jetbrains/research/ml/ast/util/Psi.kt

```diff
@@ -0,0 +1,63 @@
+package org.jetbrains.research.ml.ast.util
+
+import com.intellij.codeInsight.daemon.impl.DefaultHighlightVisitorBasedInspection
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiErrorElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.SyntaxTraverser
+import com.intellij.psi.util.PsiTreeUtil
+import com.jetbrains.python.psi.impl.references.PyOperatorReference
+import org.jetbrains.annotations.Nullable
+
+fun PsiElement.isCorrect(): Boolean {
+    if (this.hasSyntaxError() || !this.canBeResolved()) {
+        return false
+    }
+    return (this as PsiFile).countHighlightErrors() == 0
+}
+
+// Does not help find errors like unresolved reference and etc
+fun PsiElement.hasSyntaxError(): Boolean = this.countPsiErrorElements() != 0
+
+private fun PsiElement.countPsiErrorElements(): Int {
+    return ApplicationManager.getApplication().runReadAction<Int> {
+        val elements = PsiTreeUtil.collectElements(this) { it is PsiErrorElement }
+        elements.size
+    }
+}
+
+// Find unresolved references errors
+fun PsiElement.canBeResolved(): Boolean {
+    this as @Nullable PsiFile
```

Comment:
Not sure, what is this line for?

Reply:
I did not notice it, sorry. It is an old code line